src/nvim/api/private/defs.h

```diff
@@ -30,10 +30,14 @@ typedef enum {
   kMessageTypeNotification
 } MessageType;
 
-/// Used as the message ID of notifications.
+/// Used as response id of notifications. Api methods are allowed to not
+/// calculate the return value, as it will be discarded anyway.
 #define NO_RESPONSE UINT64_MAX
 
-/// Used as channel_id when the call is local
+/// Used as response id when the call is local or part of a multi-request.
+#define INTERNAL_RESPONSE (UINT64_MAX-1u)
```

Comment:
When thinking of it, no API method is actually using the response id number, nor would I expect it to be used (possibly in #4971, but then I guess it would be part of the stream, and that would need to change signatures of methods using  anyway), so we could just make this a `bool` for now, or remove it from this PR, it could be introduced when actually used.